poc/frost.sage

```diff
@@ -100,10 +101,10 @@ class Signer(object):
 
     # XXX(caw): move this out to a helper function?
     def verify_share(self, group_comm, participant_list, index, signer_key, signer_share, signer_comm, msg):
-        msg_hash = self.H3(msg)
+        # msg_hash = self.H3(msg)
```

Comment:
Re: 

>The pre-hash step introduced in https://github.com/cfrg/draft-irtf-cfrg-frost/pull/46, to address https://github.com/cfrg/draft-irtf-cfrg-frost/issues/43, meant that the vanilla ed25519 interop wouldn't work. (Vanilla ed25519 doesn't pre-hash the message when computing the challenge.) I think we got the change in https://github.com/cfrg/draft-irtf-cfrg-frost/pull/46 wrong. [The audit](https://github.com/ZcashFoundation/redjubjub/blob/main/zcash-frost-audit-report-20210323.pdf) only recommends this pre-hashing step when computing the blinding factor, not when computing the challenge, so I think we could revert the pre-hashing step for the challenge.

Reply:
Yeah, if we agree that we should revert this change, I'll tidy up this code and fix the main spec.